source/crud/crud.rst

```diff
@@ -218,6 +218,10 @@ Read
      * If specified, drivers SHOULD apply this option to both the original aggregate command and subsequent
      * getMore operations on the cursor.
      *
+     * Drivers MUST NOT specify a batchSize of zero in an aggregate command that includes an $out stage, as
+     * that will prevent the pipeline from executing. Drivers SHOULD leave the cursor.batchSize command option
```

Comment:
Out of curiosity, what's the rationale for using "SHOULD" here instead of "MUST"?

Reply:
There is technically no harm in specifying batchSize >= 1, as that ensures the initial `aggregate` command does start pipeline execution to generate at least one result document. Of course, with `$out`, we'll expect `aggregate` to return an empty batch of results and a cursor ID of zero (since there really are no result documents to return).

When using `batchSize` of zero, `aggregate` returns an empty batch (of course) with a non-zero cursor ID. That would require iteration via `getMore` (presumably with batchSize >= 1) to ensure at least one document is requested from its result, which will trigger the pipeline to execute.

Reply:
If that didn't answer the question, I was open to the fact that drivers could have passed [SPEC-826](https://jira.mongodb.org/browse/SPEC-826) simply by including a check for zero but still allowing larger batch sizes. So if drivers are currently allowing a batchSize >= 1 and that doesn't conflict with the spec test you added, they can continue to do so. Otherwise, SHOULD might as well read as MUST per our existing RFC conventions.